app/main.py

```diff
@@ -1,25 +1,96 @@
-from abc import ABC
+from abc import ABC, abstractmethod
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __get__(self, instance: any, owner: any) -> None:
+        return getattr(instance, self._name)
+
+    def __set__(self, instance: any, value: int) -> None:
+        if not (self.min_amount <= value <= self.max_amount):
+            raise ValueError(f"""{self._name} must be between
+{self.min_amount} and {self.max_amount}""")
```

Comment:
fix the code formatting, please. You can use "black" for that.

Reply:
from abc import ABC


class IntegerRange:
    def __init__(self, min_amount: int, max_amount: int) -> None:
        self.min_amount = min_amount
        self.max_amount = max_amount

    def __get__(self, instance: any, owner: any) -> None:
        return getattr(instance, self._name)

    def __set__(self, instance: any, value: int) -> None:
        if not (self.min_amount <= value <= self.max_amount):
            raise ValueError(
                f"{self._name} must be between {self.min_amount} and {self.max_amount}"
            )
        return setattr(instance, self._name, value)

    def __set_name__(self, owner: any, name: str) -> None:
        self._name = "_" + name


class Visitor:
    def __init__(self, name: str, age: int, weight: int, height: int) -> None:
        self.name = name
        self.age = age
        self.weight = weight
        self.height = height


class SlideLimitationValidator(ABC):
    def __init__(
            self,
            age_range: IntegerRange,
            weight_range: IntegerRange,
            height_range: IntegerRange
    ) -> None:
        self.age_range = age_range
        self.weight_range = weight_range
        self.height_range = height_range


class ChildrenSlideLimitationValidator(SlideLimitationValidator):
    def __init__(self) -> None:
        self.age_range = IntegerRange(4, 14)
        self.weight_range = IntegerRange(80, 120)
        self.height_range = IntegerRange(20, 50)
        super().__init__(self.age_range, self.weight_range, self.height_range)


class AdultSlideLimitationValidator(SlideLimitationValidator):
    def __init__(self) -> None:
        self.age_range = IntegerRange(14, 60)
        self.height_range = IntegerRange(120, 220)
        self.weight_range = IntegerRange(50, 120)
        super().__init__(self.age_range, self.weight_range, self.height_range)


class Slide:
    def __init__(self, name: str, limitation_class: any) -> None:
        self.age = None
        self.weight = None
        self.height = None
        self._name = name
        self.limitation_validator = limitation_class()

    def can_access(self, visitor: Visitor) -> bool:
        try:
            self.limitation_validator(visitor.age, visitor.weight, visitor.height)
            return True
        except ValueError:
            return False 
  Here's what I have. Can you help with this and explain what and how, because I'm already lost in this((